lib/index.js

```diff
@@ -351,6 +368,10 @@ class Study extends EventTarget {
       case "downgrade":
         emit(this, "change", "normal-shutdown")
         break;
+
+      // custom, study-specific reason for ending study
+      default:
```

Comment:
If the "usual end of session" is the `shutdown|upgrade|downgrade` cases, maybe that code should also be the `default` case, and this should be:
```
case "uninstall-custom-reason":
  emit(this, "uninstall-custom-reason", reason);
```
Moved above the `shutdown|upgrade|downgrade` cases.